components/timeline/TimelineDiscover.vue

```diff
@@ -6,10 +6,54 @@ const { locale: lang } = useI18n()
 const locale = getLanguageForRecs(lang.value)
 
 const recommendations: Recommendation[] = await $fetch(`/api/recommendations?locale=${locale}`)
+
+// Shorten a string to less than maxLen characters without truncating words.
+function shorten(str: string, maxLen: number): string {
+  if (str.length <= maxLen)
+    return str
+  return `${str.slice(0, str.lastIndexOf(' ', maxLen))}...`
+}
```

Comment:
Lifted this from [here](https://stackoverflow.com/questions/5454235/shorten-string-without-cutting-words-in-javascript#answer-40382963) but used `slice` instead since `substr` is deprecated

Reply:
TIL about `substr`! I've always used `substring`